packages/yourloops/lib/patient/patients.hook.tsx

```diff
@@ -58,19 +59,53 @@ export default function usePatientsProviderCustomHook(): PatientsContextResult {
   const teamIdForWhichPatientsAreFetched = useRef(null)
 
   const fetchPatients = useCallback((teamId: string = selectedTeamId) => {
-    PatientUtils.computePatients(user, teamId).then(computedPatients => {
-      setPatients(computedPatients)
-    }).catch((reason: unknown) => {
-      const message = errorTextFromException(reason)
-      alert.error(message)
-    }).finally(() => {
-      setInitialized(true)
-      setRefreshInProgress(false)
-    })
+    PatientUtils.computePatients(user, teamId)
+      .then((computedPatients: Patient[]) => {
+        setPatients(computedPatients)
+        return computedPatients
+      })
+      .catch((reason: unknown) => {
+        const message = errorTextFromException(reason)
+        alert.error(message)
+      })
+      .finally(() => {
+        setInitialized(true)
+        setRefreshInProgress(false)
+      })
+      .then(async (computedPatients: Patient[]) => {
+        if (!isUserHcp) {
+          return
+        }
+
+        await fetchPatientsMetrics(computedPatients, teamId)
+      })
     // Need to rewrite the alert component, or it triggers infinite loop...
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [user, selectedTeam])
 
+  const fetchPatientsMetrics = async (allPatients: Patient[], teamId: string = selectedTeamId): Promise<void> => {
+    const acceptedInvitePatients = allPatients.filter((patient: Patient) => patient.invitationStatus === UserInviteStatus.Accepted)
+    if (!acceptedInvitePatients.length) {
+      return
+    }
+
+    const patientIds = acceptedInvitePatients.map((patient: Patient) => patient.userid)
+
+    const patientsMetrics = await PatientApi.getPatientsMetricsForHcp(user.id, teamId, patientIds)
+
+    const updatedPatients = allPatients.map((patient: Patient) => {
+      const relatedMetrics = patientsMetrics.find((metrics: PatientMetrics) => metrics.userid === patient.userid)
+      if (!relatedMetrics) {
+        return patient
+      }
+      patient.glycemiaIndicators = relatedMetrics.glycemiaIndicators
+      patient.monitoringAlerts = relatedMetrics.monitoringAlerts
+      patient.medicalData = relatedMetrics.medicalData
+      return patient
+    })
```

Comment:
<!-- Suggestions propose improvements to the current subject. It's important to be explicit and clear on what is being suggested and why it is an improvement. Consider using patches to further communicate your intent.  -->
**🎯 suggestion (non-blocking):** ‏All this code could be defined in another function in `patient.util.ts`